disnake/utils.py

```diff
@@ -1318,6 +1322,47 @@ def search_directory(path: str) -> Iterator[str]:
             yield prefix + name
 
 
+# this is similar to pkgutil.walk_packages, but with a few modifications
+def _walk_modules(
+    paths: Iterable[str],
+    prefix: str = "",
+    ignore: Optional[Union[Iterable[str], Callable[[str], bool]]] = None,
+) -> Iterator[str]:
+    if isinstance(ignore, str):
+        raise TypeError("`ignore` must be an iterable of strings or a callable")
+
+    if isinstance(ignore, Iterable):
+        ignore_tup = tuple(ignore)
+        ignore = lambda path: path.startswith(ignore_tup)
+    # else, it's already a callable or None
+
+    seen: Set[str] = set()
+
+    for _, name, ispkg in pkgutil.iter_modules(paths, prefix):
+        if ignore and ignore(name):
+            continue
+
+        if ispkg:
```

Comment:
How about inverting the condition to save one indent level
```py
if not ispkg:
    yield name
    continue
...
```

Reply:
Sounds good. https://github.com/DisnakeDev/disnake/pull/796/commits/b008329c83ad12009c976f28bb73a132b30dc41c